### PR TITLE
No need to wrap DDP when using Fast DDP

### PR DESCRIPTION
**Pull request by ikurtchen**
# What does this PR do?

When I enable fast ddp with `--distribution_strategy fast_ddp`, the model is wrapped with `torch.nn.parallel.DistributedDataParallel` from `accelerator.prepare_model()`, this could be incorrect since fast ddp handles gradients all-reduce by itself. In this patch, it checks if it's not fast ddp before wrap with `torch.nn.parallel.DistributedDataParallel` in `accelerator`.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
  - no
- [x] Did you make sure to update the documentation with your changes?
  - no need update doc 
- [x] Did you write any new necessary tests?
  - no need new tests

### CODE REVIEW

**Review by regisss — APPROVED** [review]
Good catch!
